### ISSUE TITLE
Missing method for Intersection2 for a collection and a list

### ISSUE THREAD

**Issue by wilfwilson** [gap-system/gap#773]
In both `stable-4.8` and `master`, I encounter the error detailed below.

It is natural to intersect a D-class of a semigroup (an equivalence class of a particular relation on a semigroup) with a list which may contain elements of the semigroup. This is possible, however if the list happens to be empty, then the following error occurs:
```
gap> S := FullTransformationMonoid(2);
<full transformation monoid of degree 2>
gap> D := GreensDClassOfElement(S, IdentityTransformation);
<Green's D-class: IdentityTransformation>
gap> Intersection(D, [IdentityTransformation, Transformation([1, 1])]);
[ IdentityTransformation ]
gap> Intersection(D, []);
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `Intersection2' on 2 arguments
```
The expected behaviour would be to return the empty list `[  ]` in this case.

**Comment by markuspf** [comment]
I can't reproduce this on either `stable-4.8` or `master`. Do you have any
packages loaded?

**Comment by wilfwilson** [comment]
I encountered and reported this error with no packages loaded (except for GAPDoc 1.5.1)

**Comment by markuspf** [comment]
Could you tell me the version line of the GAP that you are using? i.e. for me
on master it is

```
GAP v4.8.3-612-gcd4afab of 2016-05-02 11:18:31 (BST)
```

**Comment by wilfwilson** [comment]
Sure.

I also get `GAP v4.8.3-612-gcd4afab of 2016-05-02 11:29:53 (BST)` in the `master` branch.

On `stable-4.8` I get `GAP v4.8.3-55-gf532f97 of 2016-05-02 11:32:38 (BST)`.

**Comment by markuspf** [comment]
Curious, I needed to move my packages away, so some autoloaded package seems to be playing with collections/lists there.

In any case, I think the proper fix is to just catch the case of an empty list in `Intersection` anyway, since once we hit an empty list, the result will be empty as well.

Maybe @stevelinton could confirm that since I think he has been rearranging this code recently.

**Comment by markuspf** [comment]
Turns out that the package `ResClasses` installs a method for this, and hence if you load `ResClasses` this bug does not occur.

**Comment by markuspf** [comment]
Or one just installs a method for `Intersection2`, pull request incoming.

**Comment by markuspf** [comment]
See #774 for my attempt at fixing this.

**Comment by markuspf** [comment]
This is now fixed after merging #774
